data/data.go

```diff
@@ -86,52 +42,86 @@ func (e Event) HumanTime() string {
 	return time.Unix(e.Time/1000, 0).Format(time.RFC1123)
 }
 
-func NewMeetupSchedule() *MeetupSchedule {
-	return &MeetupSchedule{
-		events: make(map[string][]Event),
-	}
-}
+type Schedule struct {
+	key string
 
-type MeetupSchedule struct {
-	events map[string][]Event
+	Label  string
+	Events []Event
 }
 
-func (m *MeetupSchedule) Add(name string, events []Event) {
-	m.events[name] = events
-}
+type Schedules []*Schedule
+
+func (s *Schedule) FetchEvents(client Client) (err error) {
+	url := fmt.Sprintf("https://api.meetup.com/%s/events?status=upcoming", s.key)
+
+	data, err := client.Get(url)
 
-func (m *MeetupSchedule) BoulderEvents() []Event {
-	return nextThree(m.events["Boulder-Gophers"])
+	if err != nil {
+		return err
+	}
+
+	err = json.Unmarshal(data, &s.Events)
+
+	if err != nil {
+		return err
+	}
+
+	sort.SliceStable(s.Events, func(i, j int) bool {
```

Comment:
I looked up `sort.Slice` to learn about it and saw the comment `"The sort is not guaranteed to be stable. For a stable sort, use SliceStable."`.  I reached for `SliceStable` thinking that in the case where 2 events had the same time, it would preserve the order that they were returned by the API.  I didn't know about the performance cost and the possibility of 2 events with the same timestamp seems pretty slim.  I'm fine with reverting this.

Reply:
Stable sort sounds like a good idea then. From what I recall the meetup API does not return very precise timestamps. They are rounded significantly. This increases the chance of timestamps being the same. If that is the case preserving order makes sense.

Good catch!